### ISSUE TITLE
Component: IconButton

### ISSUE THREAD

**Issue by benediktziegler** [frankdugan3/pyro#30]
**Pre-check**

This component proposal:

- [x] Is not a duplicate of an existing issue
- [x] Does not require 3rd party JS to implement
- [x] Considers mobile-friendliness
- [x] Does not cause accessibility problems

**Describe the use case for this component**
An icon button as a convenience component which wraps the button component but without rendering the button part. Possibly render a label next to the icon or as tooltip.

Similar to the Petal icon_button but `shape` could allow here to define the hover bg shape (options: rounded, square, circle) and the button's variant could be dropped for the icon_button interface.

This could be used for the theme-selector or other nav-bar items.

**Demonstrate the component's API**

```heex
<.icon_button icon_name="hero-cog-8-tooth-solid" tooltip="Settings" />
```

**Comment by frankdugan3** [comment]
Yeah, I think I like this proposal. One thing to ensure is that it fully supports functioning as a link (see the source for the btn component) as well as a button. Were you interesting in putting together a PR for this?

**Comment by benediktziegler** [comment]
Yes, I can work on that and put together a PR.
